common-template/src/main/java/com/redhat/cloud/notifications/qute/templates/TemplateService.java

```diff
@@ -0,0 +1,68 @@
+package com.redhat.cloud.notifications.qute.templates;
+
+import com.redhat.cloud.notifications.ingress.Action;
+import com.redhat.cloud.notifications.qute.templates.mapping.Console;
+import com.redhat.cloud.notifications.qute.templates.mapping.DefaultTemplates;
+import com.redhat.cloud.notifications.qute.templates.mapping.OpenShift;
+import com.redhat.cloud.notifications.qute.templates.mapping.Rhel;
+import io.quarkus.qute.Engine;
+import io.quarkus.qute.TemplateInstance;
+import jakarta.annotation.PostConstruct;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.util.HashMap;
+import java.util.Map;
+
+@ApplicationScoped
+public class TemplateService {
+
+    @Inject
+    Engine engine;
+
+    Map<TemplateDefinition, String> templatesConfigMap = new HashMap<>();
+
+    @PostConstruct
+    public void init() {
+        templatesConfigMap.putAll(DefaultTemplates.templatesMap);
+        templatesConfigMap.putAll(Console.templatesMap);
+        templatesConfigMap.putAll(Rhel.templatesMap);
+        templatesConfigMap.putAll(OpenShift.templatesMap);
+    }
+
+    public TemplateInstance compileTemplate(final TemplateDefinition config) {
+
+        String path = templatesConfigMap.get(config);
+        if (path == null) {
+            path = templatesConfigMap.get(new TemplateDefinition(config.integrationType(), config.bundle(), config.application(), null));
+        }
+        if (path == null) {
+            path = templatesConfigMap.get(new TemplateDefinition(config.integrationType(), null, null, null));
+        }
+
+        if (null != path) {
+            return engine.getTemplate(config.integrationType().name().toLowerCase() + "/" + path).instance();
+        }
```

Comment:
Please add a comment to explain we're asking Qute to load the template file from its full path (maybe provide a full path example?).